pyomo/contrib/doe/examples/reactor_kinetics.py

```diff
@@ -0,0 +1,258 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#
+#  Pyomo.DoE was produced under the Department of Energy Carbon Capture Simulation
+#  Initiative (CCSI), and is copyright (c) 2022 by the software owners:
+#  TRIAD National Security, LLC., Lawrence Livermore National Security, LLC.,
+#  Lawrence Berkeley National Laboratory, Pacific Northwest National Laboratory,
+#  Battelle Memorial Institute, University of Notre Dame,
+#  The University of Pittsburgh, The University of Texas at Austin,
+#  University of Toledo, West Virginia University, et al. All rights reserved.
+#
+#  NOTICE. This Software was developed under funding from the
+#  U.S. Department of Energy and the U.S. Government consequently retains
+#  certain rights. As such, the U.S. Government has been granted for itself
+#  and others acting on its behalf a paid-up, nonexclusive, irrevocable,
+#  worldwide license in the Software to reproduce, distribute copies to the
+#  public, prepare derivative works, and perform publicly and display
+#  publicly, and to permit other to do so.
+#  ___________________________________________________________________________
+
+
+import pyomo.environ as pyo
+from pyomo.dae import ContinuousSet, DerivativeVar
+import numpy as np
+from enum import Enum
+from pyomo.contrib.doe import model_option_lib
+
+
+def disc_for_measure(m, nfe=32, block=True):
+    """Pyomo.DAE discretization
+
+    Arguments
+    ---------
+    m: Pyomo model
+    nfe: number of finite elements
+    block: if True, the input model has blocks
+    """
+    discretizer = pyo.TransformationFactory('dae.collocation')
+    if block:
+        for s in range(len(m.block)):
+            discretizer.apply_to(m.block[s], nfe=nfe, ncp=3, wrt=m.block[s].t)
+    else:
+        discretizer.apply_to(m, nfe=nfe, ncp=3, wrt=m.t)
+    return m
+
+
+def create_model(
+    mod=None,
+    model_option=model_option_lib.stage2,
+    control_time=[0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1],
```

Comment:
If someone defines t_control=[5], they can define it; No error checks at this stage, since when we define `MeasurementVariables`, models haven't been discretized, so we don't know what timepoints the model has. The error check is after we discretize, formulate and solve the model, a function will check if this measurement is in the model; if not, a ValueError is thrown saying this measurement is not in the model.